File: python/AedatTools/ImportAedatExample.py
# -*- coding: utf-8 -*-
"""
Created on Mon May 23 11:38:50 2016

@author: sim

Example script for how to invoke the ImportAedat function
"""

import sys
#sys.path.append('/home/federico/space/inilabs/data/multicam/cam0/')

import ImportAedat

# Create a dict with which to pass in the input parameters.
args = {}

# Put the filename, including full path, in the 'file' field.
#probelms caer_out-2016-06-23_17:24:47.aedat
args['filePathAndName'] = '/home/federico/space/inilabs/data/multicam/cam1/caer_out-2016-06-22_17_03_53_new.aedat'

# Read special events and polarity events
args['dataTypes'] = {0,1,2}

# Invoke the function
output = ImportAedat.ImportAedat(args)

File: matlab/AedatTools/ImportAedatExample.m
% Example script for how to invoke the importAedat function

clear
dbstop if error

% Create a structure with which to pass in the input parameters.
input = struct;

% Put the filename, including full path, in the 'file' field.
input.filePath = 'example.aedat';


% Add any restrictions on what to read out. 
% This example limits readout to the first 1M events (aedat fileFormat 1 or 2 only):
%input.endEvent = 1e6;

%input.startEvent = 12e6;
%input.endEvent = 24e6;

% This example limits readout to a time window between 48.0 and 48.1 s:
%input.startTime = 48;
%input.endTime = 48.1;

%This example limits the read out to certain types of event only
input.dataTypes = {'polarity', 'special'};

% Working with a file where the source hasn't been declared - do this explicitly:
input.source = 'Davis240b';

% Invoke the function
output = ImportAedat(input);
